subsys/net/ip/connection.c

```diff
@@ -752,15 +753,13 @@ enum net_verdict net_conn_input(enum net_ip_protocol proto, struct net_pkt *pkt)
 	}
 #endif
 
-	if (IS_ENABLED(CONFIG_NET_DEBUG_CONN)) {
-		u16_t chksum;
-
-		if (proto == IPPROTO_TCP) {
-			chksum = NET_TCP_HDR(pkt)->chksum;
-		} else {
-			chksum = NET_UDP_HDR(pkt)->chksum;
-		}
+	if (proto == IPPROTO_TCP) {
```

Comment:
Might be a good idea to use a switch on `proto` here. This way if more protocols are added and not handled in the switch-case, a compiler warning will be generated. The way things are right now, the compiler will happily cast the header to an UDP header and read two bytes from it, if the protocol isn't TCP (and not necessarily UDP).